backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/window/descriptor/factory/standard/GenericDocumentSummaryValueProvider.java

```diff
@@ -75,9 +75,37 @@ public static GenericDocumentSummaryValueProvider of(final DocumentEntityDescrip
 			return new GenericDocumentSummaryValueProvider(fieldValuesExtractors);
 		}
 
+		fieldValuesExtractors = extractFieldNamesFromCompositeLookup(entityDescriptor);
+		if (fieldValuesExtractors != null && !fieldValuesExtractors.isEmpty())
+		{
+			return new GenericDocumentSummaryValueProvider(fieldValuesExtractors);
+		}
 		return EMPTY;
 	}
 
+	private static List<FieldValueExtractor> extractFieldNamesFromCompositeLookup(final DocumentEntityDescriptor.Builder entityDescriptor)
+	{
+		try
+		{
+			final List<DocumentFieldDescriptor.Builder> idFields = entityDescriptor.getIdFieldBuilders();
+			if (idFields.isEmpty() || idFields.size() == 1)
+			{
+				return ImmutableList.of();
+			}
+
+			return idFields.stream()
+					.filter(field -> !field.isVirtualField())
+					.map(field -> new GenericFieldValueExtractor(field.getFieldName()))
+					.collect(Collectors.toList());
```

Comment:
just pls use ImmutableList in this case because the result is going into some descriptors that are cached and i have to be sure it is going to be immutable and not some ArrayList hidden behind List interface.